blocks/bricks/attention.py

```diff
@@ -389,12 +386,10 @@ def take_glimpses_inputs(self):
                 self.state_names)
 
     @application
-    def initial_glimpses(self, name, batch_size, attended):
-        if name == "weighted_averages":
-            return tensor.zeros((batch_size, self.attended_dim))
-        elif name == "weights":
-            return tensor.zeros((batch_size, attended.shape[0]))
-        raise ValueError("Unknown glimpse name {}".format(name))
+    def initial_glimpses(self, batch_size, attended):
+        return {
+         "weighted_averages": tensor.zeros((batch_size, self.attended_dim)),
+         "weights": tensor.zeros((batch_size, attended.shape[0]))}
```

Comment:
Indeed! Let's say those were cosmic rays influencing  my brain.. I will definitely do it as you suggest.